tests/scripts/auto-grow-storage.sh

```diff
@@ -0,0 +1,217 @@
+            fi
+            echo "Total count of OSDs for deviceset ${deviceSetName} is set to ${finalCount}."
+            result=$(oc patch CephCluster rook-ceph -n "${ns}" --type json --patch "[{ "op": "replace", "path": "/spec/storage/storageClassDeviceSets/${deviceSetCount}/count", "value": ${finalCount} }]")
+            echo "${result}"
+            break
+         fi
+         deviceSetCount=$((deviceSetCount+1))    
+      done
+      currentCount=$((currentCount+1))
+   done
+}
+
+function growOSD(){
+   itr=0
+   alertmanagerroute=$(oc -n openshift-user-workload-monitoring get routes | awk '/thanos-ruler/ { print $2 }')
+   curl -sk -H "Authorization: Bearer $(oc sa get-token prometheus-k8s -n openshift-monitoring)"  https://"${alertmanagerroute}"/api/v1/alerts | jq -r '.' >./tt.txt
+   export total_alerts
+   total_alerts=$( jq '.data.Alerts | length' < tt.txt)
+   echo "Looping at $(date +"%Y-%m-%d %H:%M:%S")"
+
+   while true
+   do
+      if [ "${total_alerts}" == "" ]
+      then
+         echo "Alert manager not configured,re-run the script"
+         exit 1
+      fi
+      export entry
+      entry=$( jq ".data.Alerts[$itr]" < tt.txt)
+      thename=$(echo "${entry}" | jq -r '.labels.alertname')
+      if [ "${thename}" = "CephOSDNearFull" ] || [ "${thename}" = "CephOSDCriticallyFull" ]
+      then
+         echo "${entry}"
+         ns=$(echo "${entry}" | jq -r '.labels.namespace')
+         osdID=$(echo "${entry}" | jq -r '.labels.ceph_daemon')
+         osdID=${osdID/./-}
+         pvc=$(oc get deployment -n rook-ceph rook-ceph-"${osdID}" -o json | jq -r '.metadata.labels."ceph.rook.io/pvc"')
+         echo "Processing NearFull or Full alert for PVC ${pvc} in namespace ${ns}"
+         if [[ $1 == "count" ]]
+         then 
+            growHorizontally "$2" "${pvc}" "${ns}" "$3"
+         else
+            growVertically "$2"  "${pvc}" "${ns}" "$3"
+         fi
+      fi   
+      (( itr = itr + 1 ))
+       if [[ "${itr}" == "${total_alerts}" ]] || [[ "${total_alerts}" == "0" ]]
+      then
+         sleep 600
+         rm -f ./tt.txt
+         alertmanagerroute=$(oc -n openshift-user-workload-monitoring get routes | awk '/thanos-ruler/ { print $2 }')
+         curl -sk -H "Authorization: Bearer $(oc sa get-token prometheus-k8s -n openshift-monitoring)"  https://"${alertmanagerroute}"/api/v1/alerts | jq -r '.' >./tt.txt
+         total_alerts=$( jq '.data.Alerts | length' < tt.txt)
+         itr=0
+         echo "Looping at $(date +"%Y-%m-%d %H:%M:%S")"
+      fi  
+   done
+}
+
+function creatingPrerequisites(){
+   echo "creating Prerequisites deployments"
+   # Create and applying the cluster-monitoring-config ConfigMap object
+   echo "apiVersion: v1
+kind: ConfigMap
+metadata:
+   name: cluster-monitoring-config
+   namespace: openshift-monitoring
+data:
+   config.yaml: |
+      enableUserWorkload: true" > cluster-monitoring-config.yaml
+   oc create -f cluster-monitoring-config.yaml
+   sleep 20
```

Comment:
Do not sleep, wait with a timeout for the expected resource to be created.

Reply:
See for reference https://github.com/rook/rook/blob/master/.github/workflows/canary-integration-test.yml#L737

Reply:
Done